05/app.js

```diff
@@ -0,0 +1,50 @@
+function Student(name, surname) {
+  this.name = name;
+  this.surname = surname;
+  this.grades = {}
+}
+//METODY:
+// dodaj ocenę z przedmiotu (i przedmiot też jeśli, go nie ma)
+// pobierz średnią arytm. z przedmiotu ('argument') lub z całości ()
+//pokaż oceny z danego przedmiotu (???)
+
+
+Student.prototype.addGrades = (function (sub, ...num) {
+  if (typeof this.grades[sub] === 'undefined') {
+    console.log(`a new subject - ${sub} - added`)
+    this.grades[sub] = [];
+    this.grades[sub].push(...num)
+  } else {
+    console.log(`new grades added to subject: ${sub}`)
+    this.grades[sub].push(...num)
+  }
+})
+
+Student.prototype.getAverageGrade = (function (sub) {
+  if (typeof sub === 'undefined') {
+    let arr = [];
+    for (item in this.grades) {
+      arr.push(...(this.grades[item]))
+    }
+    return (arr.reduce((a, b) => a + b) / arr.length).toFixed(2)
+  } else if (typeof this.grades[sub] === 'undefined') {
+    console.log('This subject does not exist on the list')
+  } else if (!this.grades[sub].length > 0) {
```

Comment:
uwaga na taki zapis. łatwo się pomylić, lepiej: `!(this.grades[sub].length > 0)` lub najłatwiej -> `this.grades[sub].length === 0`